src/basic/path-util.c

```diff
@@ -947,3 +947,68 @@ bool dot_or_dot_dot(const char *path) {
 
         return path[2] == 0;
 }
+
+char *normalize_path(const char *path) {
```

Comment:
How about making `normalize_path()` take an optional root path, like `chase_symlinks()`?

Reply:
... gah.  Thinking about this again, I probably should just use chase_symlinks().

I wasn't sure if I should add symlink chasing where it wasn't before... but if I don't, I guess the _source_ directory can escape /sysroot.

Reply:
Is it possible to use `chase_symlinks()` here? It might be, but I am not sure.

Reply:
Agreed it might not be possible for it to work - and so it will not prevent every single possible weirdness.  However chase_symlinks() is already used in fstab-generator.  I think the effect is fairly reasonable, and it means it reads consistently with the other code.

It should prevent all incorrect results from literal ".."s, without me having to write a weird new function to do that.  (Oops, too late :).  Goodbye, weird function).